python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -451,22 +452,23 @@ def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
                     pattern_skip_array=np.array(
                         match_tolerance.failedPatternList)
                 )
-            if len(matcher_struct.matches) < 0 and \
-               match_tolerance.lastPattern is None and \
-               try_idx == 0:
+            if try_idx == 0 and \
+               len(matcher_struct.matches) == 0 and \
+               match_tolerance.lastMatchedPattern is not None:
                 # If we found a pattern on a previous run and can't
                 # find an optimistic match with the harshest tolerances,
                 # the match we found previously was likely bad. We append
                 # its index to a list so we do not use it in future match
                 # iterations.
```

Comment:
These comments sound really helpful but I confess I don't see how they relate to the tests above. Can you think of some way to rephrase them a bit that would help make the connection clearer?